tests/unit/result_tests.py

```diff
@@ -643,5 +645,43 @@ def test_iteration_no_data(self):
         result = Result(self.view001, startkey='ruby')
         self.assertEqual([x for x in result], [])
 
+    def test_iteration_integer_keys(self):
+        """
+        Test that iteration works as expected when keys are integer.
+        """
+        result = Result(self.view007)
+        self.assertEqual(len([x for x in result]), 100)
+
+    def test_iteration_pagination(self):
+        """
+        Test that iteration pagination works as expected.
+        """
+
+        class CallMock:
+            last_call = None
+
+            def __init__(self, outer):
+                self.outer = outer
+
+            def call(self, *args, **kwargs):
+                self.outer.assertNotEqual((args, kwargs), self.last_call,
```

Comment:
This is just asserting that the current view query isn't the same as the previous right? How about checking against a list of expected stuff, i.e.
```
call 1: {'limit': 28}
call 2: {'limit': 28, 'startkey': 1, 'startkey_docid': 'julia027'}
call 3: {'limit': 28, 'startkey': 1, 'startkey_docid': 'julia054'}
call 4: {'limit': 28, 'startkey': 1, 'startkey_docid': 'julia081'}
```

Reply:
It's indeed more strict, I like it